diff -ur binutils-2.34_orig/ld/emultempl/pep.em binutils-2.34/ld/emultempl/pep.em
--- binutils-2.34_orig/ld/emultempl/pep.em	2020-01-18 19:55:48.000000000 +0600
+++ binutils-2.34/ld/emultempl/pep.em	2021-07-04 21:13:08.291070200 +0600
@@ -18,25 +18,25 @@
 rm -f e${EMULATION_NAME}.c
 (echo;echo;echo;echo;echo)>e${EMULATION_NAME}.c # there, now line numbers match ;-)
 fragment <<EOF
-/* Copyright (C) 2006-2020 Free Software Foundation, Inc.
-   Written by Kai Tietz, OneVision Software GmbH&CoKg.
+/* 
+   
 
-   This file is part of the GNU Binutils.
+   
 
-   This program is free software; you can redistribute it and/or modify
-   it under the terms of the GNU General Public License as published by
-   the Free Software Foundation; either version 3 of the License, or
-   (at your option) any later version.
-
-   This program is distributed in the hope that it will be useful,
-   but WITHOUT ANY WARRANTY; without even the implied warranty of
-   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-   GNU General Public License for more details.
-
-   You should have received a copy of the GNU General Public License
-   along with this program; if not, write to the Free Software
-   Foundation, Inc., 51 Franklin Street - Fifth Floor, Boston,
-   MA 02110-1301, USA.  */
+   
+   
+   
+   
+
+   
+   
+   
+   
+
+   
+   
+   
+   */
 
 
 /* For WINDOWS_XP64 and higher */
