diff -ur gettext-0.21_orig/gettext-tools/woe32dll/gettextsrc-exports.c gettext-0.21/gettext-tools/woe32dll/gettextsrc-exports.c
--- gettext-0.21_orig/gettext-tools/woe32dll/gettextsrc-exports.c	2019-12-23 17:21:12.000000000 +0600
+++ gettext-0.21/gettext-tools/woe32dll/gettextsrc-exports.c	2021-06-05 16:32:41.254923200 +0600
@@ -48,6 +48,7 @@
 VARIABLE(formatstring_php)
 VARIABLE(formatstring_python)
 VARIABLE(formatstring_python_brace)
+VARIABLE(formatstring_ruby)
 VARIABLE(formatstring_qt)
 VARIABLE(formatstring_qt_plural)
 VARIABLE(formatstring_scheme)
